### ISSUE TITLE
Flag for only enabling formatting related cops

### ISSUE THREAD

**Issue by jacob-carlborg** [bbatsov/rubocop#3385]
As far as I know RuboCop is the closest thing to a formatting tool Ruby has. To make this even better I purpose to add a new flag to RuboCop. When this flag is given only the cops that RuboCop can autocorrect and are related to formatting would be enabled. It would also automatically autocorrect all offences and not report them.

**Comment by jonas054** [comment]
The best long term solution would probably be to move all formatting related cops into a new namespace called `Space` or `Format`. Then it would just be a matter of running
```
rubocop --auto-correct --only Space
```
But that would be a bit painful at this stage, changing the namespace for so many cops.

A less intrusive solution is to let users who want to use RuboCop as a formatter do something like this:

```
rubocop --show-cops | egrep '^\S+(Space|Indent|Align)' | awk 'BEGIN { print "AllCops:\n  DisabledByDefault: true" } /./ {print $0 "\n  Enabled: true"} /^Lint/ {print "  AutoCorrect: true"}' > space_cops.yml
rubocop --auto-correct --config space_cops.yml
```

**Comment by jacob-carlborg** [comment]
Would it be possible to have a cop in multiple namespaces?

**Comment by jonas054** [comment]
No, that would be a duplicate. Not sure why you ask.

**Comment by jacob-carlborg** [comment]
> But that would be a bit painful at this stage, changing the namespace for so many cops.

I assumed for backwards compatible reasons.

**Comment by bbatsov** [comment]
I think having a "Format"/"Layout" namespace is not a bad idea at all.

**Comment by backus** [comment]
> I think having a "Format"/"Layout" namespace is not a bad idea at all.

@bbatsov This could be rolled into something like I proposed here: https://github.com/bbatsov/rubocop/issues/3414#issuecomment-239661264

Basically, some internal RuboCop reorganization + CLI additions could treat categories of cops just like third part cops (so `Lint`, `Metrics`, `Performance`, `Rails`, `Style`, `Formatting`, `RSpec`) and the proposed CLI change could make this easy to work with `--use lint,metrics,formatting` etc.

**Comment by bbatsov** [comment]
Even now you can run only a specific cop category using `--only`. Treating built-in cops as loadable add-ons doesn't make sense to me. I'd say loading something is different from running something.

**Comment by backus** [comment]
Good point I didn't know about `--only`

**Comment by jacob-carlborg** [comment]
@jonas054 thanks.
